tests/test_show_diff_format.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+
+def test_show_diff_format_none_json(cli_runner, data_archive):
+    # Check that the json output is a commit metadata only
+    with data_archive("points.tgz"):
+        r = cli_runner.invoke(["show", "--diff-format=none", "-o", "json"])
+        output = json.loads(r.stdout)
+
+        # Assert that the output is a dict with a key for metadata only
+        assert output.keys() == {"kart.show/v1"}
+
+
+def test_show_diff_format_no_data_changes_json(cli_runner, data_archive):
+    # Check that the json output is a commit metadata (if any) and a boolean for data_changes (feature/tile changes)
+    with data_archive("points.tgz"):
+        r = cli_runner.invoke(["show", "--diff-format=no-data-changes", "-o", "json"])
+        output = json.loads(r.stdout)
+        assert output["kart.diff/v1+hexwkb"] == {
+            "nz_pa_points_topo_150k": {"data_changes": True}
+        }
+
+
+def test_show_diff_format_none_text(cli_runner, data_archive):
+    # Check that the text output is a commit metadata only
+    with data_archive("points.tgz"):
+        r = cli_runner.invoke(["show", "--diff-format=none", "-o", "text"])
+        output = r.stdout
```

Comment:
Use r.stdout.splitlines() - then the expected output gets to be formatted better, instead of being on one long line with "\n" literals in it